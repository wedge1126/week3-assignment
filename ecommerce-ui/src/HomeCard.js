import React from 'react';
import PropTypes from 'prop-types';

import Rating from './Rating';

export default class HomeCard extends React.Component {
  render() {
    const { home, active, onAddToCart } = this.props;
    const currencyFormat = new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'USD',
        minimumFractionDigits: 0,
        maximumFractionDigits: 0
    });

    return <div className="home-card">
        <img className="home-card-image" src={home.image} alt={home.title} />
        <div className="home-card-body">
            <div className="home-card-type">
                <span>{home.houseType}</span> &#8226; <span>{home.location.city}, {home.location.country}</span>
            </div>
            <div className="home-card-title">{home.title}</div>
            <div>
                <span className="home-card-cost">{currencyFormat.format(home.payment.cost)}/night</span>
                <span>{home.payment.description}</span>
            </div>
            <Rating value={home.rating} />
            <div>
                <span className="host-name">{home.host.name}</span>
                <span className="superhost">{home.host.isSuperhost ? 'Superhost' : ''}</span>
            </div>
            <div>
                <button className="home-card-button" disabled={!active} onClick={() => onAddToCart(home)} >Book</button>
            </div>
        </div>
    </div>
  }
}

HomeCard.propTypes = {
    home: PropTypes.shape({
        title: PropTypes.string.isRequired,
        houseType: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
        location: PropTypes.shape({
            city: PropTypes.string.isRequired,
            country: PropTypes.string.isRequired
        }).isRequired,
        payment: PropTypes.shape({
            cost: PropTypes.number.isRequired,
            description: PropTypes.string.isRequired
        }).isRequired,
        host: PropTypes.shape({
            name: PropTypes.string.isRequired,
            isSuperhost: PropTypes.bool.isRequired
        }).isRequired,
        rating: PropTypes.shape({
            stars: PropTypes.number.isRequired,
            reviews: PropTypes.number.isRequired
        })
    }).isRequired,
    active: PropTypes.bool.isRequired,
    onAddToCart: PropTypes.func.isRequired
}