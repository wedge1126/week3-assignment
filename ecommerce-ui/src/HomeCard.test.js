import React from 'react';
import { shallow, mount } from 'enzyme';

import HomeCard from './HomeCard';

const testHome = {
    "title": "Quaint 31,400 kmÂ³ home above lovely forest",
    "houseType": "Entire Space Station",
    "image": "http://www.islandbreath.org/2015Year/02/150205deathstar.jpg",
    "location": {
        "city": "Endor",
        "country": "Endor System"
    },
    "payment": {
        "cost": 3200000000,
        "description": ""
    },
    "host": {
        "name": "Sheev",
        "isSuperhost": true
    },
    "rating": {
        "stars": 3.2,
        "reviews": 501
    }
};

describe('HomeCard', () => {
    it('should render the test data correctly', () => {
        const active = true;
        const onAddToCart = jest.fn();
        const component = shallow(<HomeCard home={testHome} active={active} onAddToCart={onAddToCart} />);
        
        expect(component).toMatchSnapshot();
    });

    it('should call onAddToCart when button is clicked', () => {
        const active = true;
        const onAddToCart = jest.fn();
        const component = mount(<HomeCard home={testHome} active={active} onAddToCart={onAddToCart} />);
        
        component.find('button.home-card-button').simulate('click');

        expect(onAddToCart).toHaveBeenCalled();

        component.unmount();
    });

    it('should not call onAddToCart when button is disabled', () => {
        const active = false;
        const onAddToCart = jest.fn();
        const component = mount(<HomeCard home={testHome} active={active} onAddToCart={onAddToCart} />);
        
        component.find('button.home-card-button').simulate('click');

        expect(onAddToCart).not.toHaveBeenCalled();
        component.unmount();
    });
});