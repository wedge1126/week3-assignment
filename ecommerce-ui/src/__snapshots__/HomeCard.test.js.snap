// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCard should render the test data correctly 1`] = `
<div
  className="home-card"
>
  <img
    alt="Quaint 31,400 km³ home above lovely forest"
    className="home-card-image"
    src="http://www.islandbreath.org/2015Year/02/150205deathstar.jpg"
  />
  <div
    className="home-card-body"
  >
    <div
      className="home-card-type"
    >
      <span>
        Entire Space Station
      </span>
       • 
      <span>
        Endor
        , 
        Endor System
      </span>
    </div>
    <div
      className="home-card-title"
    >
      Quaint 31,400 km³ home above lovely forest
    </div>
    <div>
      <span
        className="home-card-cost"
      >
        $3,200,000,000
        /night
      </span>
      <span />
    </div>
    <HomeCard
      value={
        Object {
          "reviews": 501,
          "stars": 3.2,
        }
      }
    />
    <div>
      <span
        className="host-name"
      >
        Sheev
      </span>
      <span
        className="superhost"
      >
        Superhost
      </span>
    </div>
    <div>
      <button
        className="home-card-button"
        disabled={false}
        onClick={[Function]}
      >
        Book
      </button>
    </div>
  </div>
</div>
`;
